include/slate/internal/MatrixStorage.hh

```diff
@@ -1111,35 +1198,7 @@ template <typename scalar_t>
 TileInstance<scalar_t>& MatrixStorage<scalar_t>::tileAcquire(
```

Comment:
Do we need `tileAcquire`? The description always seemed vague to me what its purpose was, different than some other routine like `tileInsert` or `tileGetFor___`. The way it is used was to allocate a tile without copying contents from an existing valid tile, but then needed `tileModified` to set the MOSI status; see [src/internal/internal_gecopy.cc](https://github.com/icl-utk-edu/slate/blob/c215fe23e4fe23a9c67137a79fd0cb8ea7d2f915/src/internal/internal_gecopy.cc#L258-L261). I always argued that requiring 2 function calls (`tileAcquire` and later `tileModified`) for correctness was a bad design.

Probably a longer discussion for future cleanup.

Reply:
It looks like `MatrixStorage::tileAcquire` is only called from `BaseMatrix::tileAcquire` and `BaseMatrix::tileGet` and isn't user accessible, so I guess I might as well manually inline those calls and remove the function.

For the `BaseMatrix` level, I agree that needing 2 calls is error-prone and should be replace with `tileGetForOverwrite` or something.